src/Datagrid/ProxyQuery.php

```diff
@@ -79,6 +86,29 @@ public function __clone()
         $this->queryBuilder = clone $this->queryBuilder;
     }
 
+    /**
+     * Optimize query with a lot rows.
+     * Not recommended set "false" with left joins.
+     *
+     * @param bool $distinct
+     *
+     * @return ProxyQuery
+     */
+    final public function setDistinct($distinct)
+    {
+        $this->distinct = (bool) $distinct;
```

Comment:
Please throw an `\InvalidArgumentException` if `$distinct` is not a boolean. We are going to add a `bool` in next version, which means this will result in an error, which will mean a BC-break.

Reply:
Alternatively, you can just remove the argument and rename this to `disableDistinct`